Terminal.Gui/Clipboard/IClipboard.cs

```diff
@@ -1,40 +1,26 @@
-﻿using System;
+﻿namespace Terminal.Gui;
 
-namespace Terminal.Gui {
-	/// <summary>
-	/// Definition to interact with the OS clipboard.
-	/// </summary>
-	public interface IClipboard {
-		/// <summary>
-		/// Returns true if the environmental dependencies are in place to interact with the OS clipboard.
-		/// </summary>
-		bool IsSupported { get; }
+/// <summary>Definition to interact with the OS clipboard.</summary>
+public interface IClipboard {
+    /// <summary>Returns true if the environmental dependencies are in place to interact with the OS clipboard.</summary>
+    bool IsSupported { get; }
```

Comment:
(out of scope)

I realized something pretty obvious recently that I'm sure everyone implicitly already knows but which I just wanted to ensure awareness of...

Remember that an interface is a minimum contract. Implementers can define _more_ than the interface specifies, just not less.

So, an implementer could make a setter for IsSupported.

**It's their own fault**, but, if our code which uses the interface isn't at least written with that in mind, problems could occur.

That also should usually be part of the decision-making process around whether to put something in an interface, a base class, or both. If it's something that has a narrow definition that isn't likely to be different in any implementation, a base class might be a good idea.

OR, since c# 8, interfaces have now had the default implementation feature, which is pretty rad. With some minor restrictions, you can define an implementation for an interface member which will be used if an implementor does not explicitly provide their own. That could be handy for situations like this one, where this member is supposed to basically provide some sort of guarantee of underlying support for the interface's functionality. Consumers wishing to override that still can, by writing the property on their class as they otherwise would have, but it makes the default be at least mostly reliable, if it can be done in a simple way that covers a broad range of typical cases.